pkgs/servers/openvscode-server/default.nix

```diff
@@ -44,74 +35,67 @@ let
-    sed -i 's/${version}/${esbuild'.version}/g' ${path}/node_modules/esbuild/lib/main.js
-    ln -s -f ${esbuild'}/bin/esbuild ${path}/node_modules/esbuild/bin/esbuild
-  '';
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "openvscode-server";
-  version = "1.88.1";
+  version = "1.96.4";
 
   src = fetchFromGitHub {
     owner = "gitpod-io";
     repo = "openvscode-server";
     rev = "openvscode-server-v${finalAttrs.version}";
-    hash = "sha256-Yc16L13Z8AmsGoSFbvy+4+KBdHxvqLMwZLeU2/dAQVU=";
+    hash = "sha256-TUM2R3TT4xZ1BTsZaBRbq2E8/uPazKryEoyBenMJwzE=";
   };
 
-  yarnCache = stdenv.mkDerivation {
-    name = "${finalAttrs.pname}-${finalAttrs.version}-${system}-yarn-cache";
-    inherit (finalAttrs) src;
-    nativeBuildInputs = [
-      cacert
-      yarn'
-      git
-    ];
-    buildPhase = ''
-      export HOME=$PWD
-
-      yarn config set yarn-offline-mirror $out
-      find "$PWD" -name "yarn.lock" -printf "%h\n" | \
-        xargs -I {} yarn --cwd {} \
-          --frozen-lockfile --ignore-scripts --ignore-platform \
-          --ignore-engines --no-progress --non-interactive
-    '';
-
-    installPhase = ''
-      echo yarnCache
-    '';
-
-    outputHashMode = "recursive";
-    outputHashAlgo = "sha256";
-    outputHash = "sha256-89c6GYLT2RzHqwxBKegYqB6g5rEJ6/nH53cnfV7b0Ts=";
-  };
+  ## fetchNpmDeps doesn't correctly process git dependencies
+  ## presumably because of https://github.com/npm/cli/issues/5170
+  ## therefore, we're fetching all the node_module folders into
+  ## a single FOD, and unpack it in configurePhase
+  nodeModules =
+    runCommand "openvscode-server-node-modules"
+      {
+        inherit (finalAttrs) src nativeBuildInputs;
+        outputHashMode = "recursive";
+        outputHashAlgo = "sha256";
+        outputHash = "sha256-Zdk8obKjQnLkPyQ17F44YQ1LkK4YHgbhXeom5Vf5M8g=";
+      }
+      ''
+        runPhase unpackPhase
+        export HOME=$TMPDIR/home
+        npm config set progress false
+        npm config set cafile ${cacert}/etc/ssl/certs/ca-bundle.crt
+        for p in $(find -name package.json -exec dirname {} \;)
+        do (
+          set -x
+          cd $p
+          if [ -e node_modules ]
+          then
+            echo >&2 "File exists $p/node_modules"
+            exit 0
+          fi
+          mkdir node_modules
+          if [ -e package-lock.json ]
+          then npm ci --ignore-scripts --log-level=verbose
+          else npm i --ignore-scripts --log-level=verbose
+          fi
+          mkdir -p $out/$p
+          cp -r node_modules $out/$p/
+        )
+        done
+      '';
+
+  npm_config_nodedir = nodejs;
+  npm_config_node_gyp = "${nodejs}/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js";
+
+  NODE_OPTIONS = "--openssl-legacy-provider";
+  NODE_ENV = "production";
+  PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD = "1";
+  ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
```

Comment:
And I'm not familiar with those but I guess upstream reads these. In anycase it'd be nice to put a comment for these.